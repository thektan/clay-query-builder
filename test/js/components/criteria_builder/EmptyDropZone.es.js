import React from 'react';
import EmptyDropZone from 'components/criteria_builder/EmptyDropZone.es';
import {cleanup, render} from 'react-testing-library';

const connectDnd = jest.fn(el => el);

describe(
	'EmptyDropZone',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const OriginalEmptyDropZone = EmptyDropZone.DecoratedComponent;

				const {asFragment} = render(
					<OriginalEmptyDropZone
						connectDropTarget={connectDnd}
						onCriterionAdd={jest.fn()}
					/>
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);
	}
);