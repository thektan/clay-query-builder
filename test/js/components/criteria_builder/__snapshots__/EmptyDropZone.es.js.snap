// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyDropZone should render 1`] = `
<DocumentFragment>
  <div
    class="empty-drop-zone-root"
  >
    <div
      class="empty-drop-zone-target"
    >
      <div
        class="empty-drop-zone-indicator"
      />
      <div
        class="empty-drop-zone-help-message"
      >
        <div
          class="message-item"
        >
          <img
            class="message-icon"
            src="/drag-and-drop.svg"
          />
          <span>
            drag-and-drop-criterion-from-the-right-to-add-rules
          </span>
        </div>
        <div
          class="message-item"
        >
          <img
            class="message-icon"
            src="/drag-over.svg"
          />
          <span>
            drag-and-drop-over-an-existing-criteria-to-form-groups
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
